ui/src/components/overview/MobileClientCardView.js

```diff
@@ -37,67 +28,83 @@ class MobileClientCardView extends Component {
     }
   }
 
-  removeFilter(filter) {
-    this.setState({filter: ""});
+  setFilter(filterValue) {
+    this.setState({ filter: filterValue });
   }
 
   getEmptyState() {
     return (
       <EmptyState>
         <EmptyStateTitle>
-          You have no mobile apps right now. Create one to get started.
+          You have no mobile apps right now. Create one to get started;
         </EmptyStateTitle>
         <EmptyStateAction>
-          <CreateClient createButtonSize="large" />
+          {this.props.children}
         </EmptyStateAction>
       </EmptyState>
-    )
+    );
+  }
+
+  removeFilter() {
+    this.setState({ filter: '' });
+  }
+
+  updateCurrentValue(event) {
+    this.setState({ currentValue: event.target.value });
   }
 
   renderAppCards() {
-    const {mobileClients} = this.props;
+    const { mobileClients } = this.props;
     return (
       <CardGrid matchHeight fluid>
         <CardGrid.Row key={1}>
-        { mobileClients.map((app, i) => (
-            <MobileClientCardViewItem key={i} app={app} services={mockServices} builds={mockBuilds} />
-          )) }
+          { mobileClients.map((app) => {
+            const { metadata: { name: clientAppName } } = app;
+            const { filter } = this.state;
+            return clientAppName.indexOf(filter) > -1
+              ?
+                <div className="col-xs-12 col-sm-6 col-md-4 col-lg-3" key={clientAppName}>
+                  <MobileClientCardViewItem app={app} services={mockServices} builds={mockBuilds} />
+                </div>
+              :
+                null;
```

Comment:
if there is nothing matches the filter, can we show a message saying something like "No apps match the filter"?

Reply:
sure, that's a good idea